src/main/java/at/hannibal2/skyhanni/features/event/spook/TheGreatSpook.kt

```diff
@@ -44,20 +52,76 @@ object TheGreatSpook {
-        if (isFearStatEnabled()) {
+        if (!LorenzUtils.inSkyBlock) return
+
+        if (config.primalFearTimer) config.positionTimer.renderString(displayTimer, posLabel = "Primal Fear Timer")
+        if (config.fearStatDisplay) {
             SkyblockStat.FEAR.displayValue?.let {
                 config.positionFear.renderString(it, posLabel = "Fear Stat Display")
             }
         }
-        if (isTimeLeftEnabled()) config.positionTimeLeft.renderString(displayTimeLeft, posLabel = "Time Left Display")
+        if (config.greatSpookTimeLeft) config.positionTimeLeft.renderString(displayTimeLeft, posLabel = "Time Left Display")
+    }
+
+    /**
+     * REGEX-TEST: §d§lQUICK MATHS! §r§7Solve: §r§e(10*2)+12*5
+     */
+    private val mathFearMessagePattern by RepoPattern.pattern(
+        "chat.math",
+        "§d§lQUICK MATHS! §r§7Solve: §r§e(?<math>.*)",
+    )
+
+    /**
+     * REGEX-TEST: §4[FEAR] Public Speaking Demon§r§f: Speak PlasticEating!
+     */
+    private val speakingFearMessagePattern by RepoPattern.pattern(
+        "chat.speaking",
+        "§4\\[FEAR] Public Speaking Demon§r§f: (Speak|Say something interesting) (?<name>.*)!",
+    )
+
+    private fun mathSolver(math: String?) {
+        val result = math?.let { NEUCalculator.calculateOrNull(it)?.toInt() } ?: run {
+            ChatUtils.userError("Failed to solve $math!")
+            return
+        }
+        ChatUtils.clickToActionOrDisable(
+            "The result is: $result",
+            config.primalFearSolver::math,
+            actionName = "Send the result",
+            action = {
+                HypixelCommands.allChat(result.toString())
+            },
+        )
     }
 
-    private fun isTimerEnabled(): Boolean = LorenzUtils.inSkyBlock && config.primalFearTimer
+    private fun publicSpeakingSolver() {
+        ChatUtils.clickToActionOrDisable(
+            "Click to send a random string to complete the Primal Fear",
+            config.primalFearSolver::publicSpeaking,
+            actionName = "send a random string.",
+            action = {
+                HypixelCommands.allChat("I looove SkyHanni! ${StringUtils.generateRandomString(4)}")
+            },
+        )
+    }
 
-    private fun isNotificationEnabled(): Boolean = LorenzUtils.inSkyBlock && config.primalFearNotification
-    private fun isFearStatEnabled(): Boolean = LorenzUtils.inSkyBlock && config.fearStatDisplay
-    private fun isTimeLeftEnabled(): Boolean = LorenzUtils.inSkyBlock && config.greatSpookTimeLeft
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        if (!LorenzUtils.inSkyBlock) return
 
-    private fun isAllDisabled(): Boolean = !isTimeLeftEnabled() && !isTimerEnabled() && !isFearStatEnabled() && !isNotificationEnabled()
+        if (config.primalFearSolver.math) {
+            val math = mathFearMessagePattern.matchMatcher(event.message) { group("math") } ?: return
```

Comment:
since the logic is moved into mathSolver, the explicit variable math can be removed, reverted to the state above, something like: mathsolver(group("math")))